tests/network/openvpn_server.pm

```diff
@@ -0,0 +1,105 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test OpenVPN on two machines - this one is server.
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use base 'consoletest';
+use testapi;
+use lockapi;
+use mm_network;
+use mmapi 'wait_for_children';
+use utils qw(systemctl zypper_call exec_and_insert_password);
+use strict;
+
+sub run {
+    my ($self) = @_;
+    barrier_create('OPENVPN_STATIC_START',    2);
+    barrier_create('OPENVPN_STATIC_STARTED',  2);
+    barrier_create('OPENVPN_STATIC_FINISHED', 2);
+    barrier_create('OPENVPN_CA_START',        2);
+    barrier_create('OPENVPN_CA_STARTED',      2);
+    barrier_create('OPENVPN_CA_FINISHED',     2);
+    select_console "root-console";
+
+    # Configure static network, disable firewall
+    configure_default_gateway;
+    systemctl 'stop ' . $self->firewall;
+    configure_static_ip('10.0.2.101/24');
+    configure_static_dns(get_host_resolv_conf());
+
+    # Install openvpn, generate static key
+    zypper_call('in openvpn');
+    assert_script_run('cd /etc/openvpn');
+    assert_script_run('openvpn --genkey --secret static.key');
+    mutex_create 'OPENVPN_STATIC_KEY';
+
+    # Write the server config
+    assert_script_run(qq(echo "dev tun
+ifconfig 10.8.0.1 10.8.0.2
+secret /etc/openvpn/static.key" > static.conf));
+
+    # Start the server
+    systemctl('start openvpn@static');
+    systemctl('status openvpn@static -l');
+    barrier_wait 'OPENVPN_STATIC_START';
+
+    # Test the connection when also the client is ready
+    barrier_wait 'OPENVPN_STATIC_STARTED';
+    assert_script_run("ping -c5 -W1 -I tun0 10.8.0.2");
+
+    # Stop the server when also client is done
+    barrier_wait 'OPENVPN_STATIC_FINISHED';
+    systemctl('stop openvpn@static');
+
+    assert_script_run("curl -Lo easy-rsa.tgz https://github.com/OpenVPN/easy-rsa/releases/download/v3.0.5/EasyRSA-nix-3.0.5.tgz");
```

Comment:
this codebase main purpose is to cover with tests SLE / opensuse. I am not big expert in openvpn but with help of google I got that easyrsa is tool to manage PKI CA and Yast has module to do the same . also I am sure that there are some other SLE / openSUSE specific ways to achieve the same. So what your reasons to use easyrsa ? Also if you don't want to bother with Yast module maybe we can export some existing certificates ?

Reply:
Well that's true that I don't want to bother with Yast right now but it can be nice improvement in the future. The openVPN upstream recommended method is their easyrsa script. By injecting certificates our test will become red after few years. Is it big issue to use the easyrsa script for now?

Reply:
@okurz , @rwx788  WDYT ?

Reply:
easy-rsa is using openssl, valid!

Reply:
Isnt't it packaged for any of SUSE/openSUSE distributions? As we might request that or even package ourselves. That would be ideal. The only concerns regarding this approach I have are: 1) we rely on external resource, it's same like when we used some external html5 web site for the firefox tests 2) secondly we stick to single version which might not reflect current state of openssl as is outdated. I don't see an issue to try it out and see if it becomes hard to maintain and then rethink our approach. 

Update: I've checked on my TW, easy-rsa package is there.

Reply:
@dzedro for me it is not about how good or bad easy-rsa is , it is about testing methodology. More we using something not from openSUSE/SLE repositories more we testing not openSUSE/SLE but something different. 
@pdostal choosing between test which needs support every few years but covering openSUSE/SLE and test  which no need support for few years but using some custom logic not officially supported by openSUSE/SLE I would select first one. Also I hardly believe in existence of test case which you can run for few years without changes. 

>  The openVPN upstream recommended method is their easyrsa script
This sentence requires special clarification. When I will setup some production openVPN server I will for sure use upstream recommendations but when writing test case for openSUSE/SLE I don't care at all what upstream is thinking I just care how openSUSE/SLE developers  recommend meto use  openVPN because I am testing not openVPN per ce , I am testing openVPN **package** in openSUSE/SLE. 

P.S. all what I said it is just my personal opinion. And we have a simple +1 rule here in this  repo I will have no hard feelings if you will find someone else with +1 who will merge this PR in current state

Reply:
easy-rsa is part of QA:HEAD and QA:<VERSION> repo in IBS, so it can be used across all SLE12+ tests

Reply:
@czerw thanks a lot for being proactive, very much appreciated!

Reply:
Thank you very much guys! I think it's resolved now: [server](http://pdostal-server.suse.cz/tests/230) [client](http://pdostal-server.suse.cz/tests/231)